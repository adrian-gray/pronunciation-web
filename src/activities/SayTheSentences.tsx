import React from "react";

import SplitHilite from "./../components/SplitHilite";

export default (props: { ipa: string; sentences: string[] }) => {
  const { ipa, sentences } = props;

  const lines = sentences.map((line: string, index: number) => (
    <div className="sub-canvas personal-space" key={index}>
      <p className="no-margin-bottom">{line}</p>
    </div>
  ));

  return (
    <div className="headspace">
      <h3>
        {"Say the "}
        <SplitHilite str={ipa} />
        {" sentences"}
      </h3>
      {lines}
    </div>
  );
};
