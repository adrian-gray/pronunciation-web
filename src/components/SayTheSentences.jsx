import React from "react";

import SplitHilite from "./SplitHilite";

import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { withTheme } from "@material-ui/styles";

const useStyles = makeStyles(theme => ({
  headspace: theme.headspace,
  largeText: theme.largeText,
  personalSpace: {
    padding: "1em",
    margin: "1em"
  }
}));

function TongueTwisters(props) {
  const classes = useStyles(props);
  const { ipa, sentences } = props;

  const lines = sentences.map((line, index) => (
    <Paper className={classes.personalSpace} key={index}>
      <Typography>{line}</Typography>
    </Paper>
  ));

  return (
    <div className={classes.personalSpace}>
      <Typography variant="h5" gutterBottom>
        {"Say the "}
        <SplitHilite str={ipa} />
        {" sentences"}
      </Typography>
      {lines}
    </div>
  );
}

export default withTheme(TongueTwisters);
