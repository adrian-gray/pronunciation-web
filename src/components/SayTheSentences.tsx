import React from "react";

import SplitHilite from "./SplitHilite";

import Container from "react-bootstrap/Container";

export default (props: { ipa: string; sentences: string[] }) => {
  const { ipa, sentences } = props;

  const lines = sentences.map((line: string, index: number) => (
    <Container className="sub-canvas personal-space" key={index}>
      <p className="no-margin-bottom">{line}</p>
    </Container>
  ));

  return (
    <div className="personal-space">
      <h3>
        {"Say the "}
        <SplitHilite str={ipa} />
        {" sentences"}
      </h3>
      {lines}
    </div>
  );
};
