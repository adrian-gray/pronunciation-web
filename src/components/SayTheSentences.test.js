import React from "react";
import { render, cleanup } from "@testing-library/react";

import SayTheSentences from "./SayTheSentences";

afterEach(cleanup);

test("SayTheSentences", async () => {
  const props = {
    ipa: "c~a~t",
    sentences: ["my ~hat~ has ears", "My ~ca~t does not"]
  };

  const { container } = render(<SayTheSentences {...props} />);

  expect(container).toMatchSnapshot();
});
