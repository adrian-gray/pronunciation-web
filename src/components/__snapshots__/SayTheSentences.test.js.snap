// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SayTheSentences 1`] = `
<div>
  <div
    class="makeStyles-personalSpace-3"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Say the 
      <span>
        c
      </span>
      <span
        class="makeStyles-hilite-34"
      >
        a
      </span>
      <span>
        t
      </span>
       
       sentences
    </h5>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-personalSpace-3 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        my ~hat~ has ears
      </p>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-personalSpace-3 MuiPaper-rounded"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        My ~ca~t does not
      </p>
    </div>
  </div>
</div>
`;
